spock-core/src/main/java/org/spockframework/runtime/NamespacedExtensionStore.java

```diff
@@ -0,0 +1,61 @@
+package org.spockframework.runtime;
+
+import org.junit.platform.engine.support.store.NamespacedHierarchicalStore;
+import org.junit.platform.engine.support.store.NamespacedHierarchicalStoreException;
+import org.spockframework.runtime.extension.IStore;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class NamespacedExtensionStore implements IStore {
+  private final NamespacedHierarchicalStore<Namespace> delegate;
+  private final IStore.Namespace namespace;
+
+  public NamespacedExtensionStore(NamespacedHierarchicalStore<Namespace> delegate, Namespace namespace) {
+    this.delegate = delegate;
+    this.namespace = namespace;
+  }
+
+  @Override
+  public Object get(Object key) {
+    return execute(() -> delegate.get(namespace, key));
+  }
+
+  @Override
+  public <V> V get(Object key, Class<V> requiredType) {
+    return execute(() -> delegate.get(namespace, key, requiredType));
+  }
+
+  @Override
+  public <K, V> Object getOrComputeIfAbsent(K key, Function<K, V> defaultCreator) {
```

Comment:
```suggestion
  public <K, V> V getOrComputeIfAbsent(K key, Function<K, V> defaultCreator) {
```

Reply:
I would think this should be `V`, when the user had already provided the function mapping.
Or are you concerned about the potential `ClassCastEx`, when another API was used?